### ISSUE TITLE
Checkboxes?

### ISSUE THREAD

**Issue by john-landgrave** [gingko/client#63]
I realize that there are still features forthcoming, but a definite "must" for my workflow is checkboxes, and I would love if the web interface's [ ] and [X] notation would create the checkboxes in the desktop app as well?

If you could at least point to where/how any pre-existing text transformation code exists, I'd be happy to give it a shot and submit a PR.

**Comment by AdrianoFerrari** [comment]
Hi John.

I haven't thought about how I'd go about implementing this yet, but my first impulse is to use Regex to replace `[ ]` with some html, and in those html tags create an onClick handler that sends a message back to Elm with some identifier for which checkbox was checked (i.e. index of the match).

https://github.com/gingko/client/blob/master/src/elm/Trees.elm#L565

If you decide to try it, feel free to ask me questions any time in the process.
I love learning Elm and helping others do the same, so don't hesitate :smile:

**Comment by AdrianoFerrari** [comment]
Something like this should work, @john-landgrave :
```elm
    processedContent =
      content
--     ...
        |> Regex.replace Regex.All
            (Regex.regex "\\[ \\] ")
            (\{number} -> "<input type='checkbox' onClick='checkboxClick(" ++ cardId ++ "," ++ (toString number) ++ ")'/>"  )
```

You'll also need to pass the cardId to that onClick handler, which means that `viewContent` needs to also accept the cardId. It's new type should be:
```elm
viewContent : String -> String -> Html Msg
viewContent cardId content =
--  function body here
```

You'll then need to:
[ ] Implement `checkBoxClick` in `main.js`, and have it send a message using `toElm`.
[ ] Add a message in `IncomingMsg` called `CheckboxClicked String Int` and handle it in Elm.

Sounds like a lot of changes, but the compiler (and I), can help at each step.

**Comment by 8bfcpcch** [comment]
I'm glad I found this, as I was going to report it as a bug. I guess the problem for me comes down to documentation and help. Whereas there is a plethora of information on the app website, there is virtually nothing on the client help regarding how to use the app and format cards (even if just a reference to the app website faqs). I assumed checkboxes were part of the markdown language and since things like headings # and bold ** worked, that checkboxes would as well. 

I think you need checkboxes to make this a viable client comparable to the app. But if it can't be there, then it should at least be clearly stated how to format cards and what does and doesn't work compared to the app.

Speaking of checkboxes, I was going to push the envelop and ask for some sort of command to uncheck all checkboxes in selected cards or something. This might be useful for something like daily tasks. Probably not possible, but worth a mention.

**Comment by AdrianoFerrari** [comment]
Implemented in [v2.4.6](https://github.com/gingko/client/releases/tag/v2.4.6).
